samples/Common.c

```diff
@@ -148,13 +150,15 @@ STATUS masterMessageReceived(UINT64 customData, PReceivedSignalingMessage pRecei
         CHK_STATUS(createSampleStreamingSession(pSampleConfiguration, pReceivedSignalingMessage->signalingMessage.peerClientId, TRUE,
                                                 &pSampleStreamingSession));
         pSampleStreamingSession->firstSdpMsgReceiveTime = GETTIME();
-        pSampleConfiguration->sampleStreamingSessionList[pSampleConfiguration->streamingSessionCount++] = pSampleStreamingSession;
+        pSampleConfiguration->sampleStreamingSessionList[pSampleConfiguration->streamingSessionCount] = pSampleStreamingSession;
+        ATOMIC_INCREMENT(&pSampleConfiguration->streamingSessionCount);
```

Comment:
This one is not atomic. Could you check with Felix whether this is already protected by the object lock and no need to provide for atomicity.